aws/resource_aws_emr_cluster.go

```diff
@@ -666,6 +685,10 @@ func resourceAwsEMRClusterRead(d *schema.ResourceData, meta interface{}) error {
 		log.Printf("[ERR] Error setting EMR configurations for cluster (%s): %s", d.Id(), err)
 	}
 
+	if err := d.Set("configurations_json", cluster.Configurations); err != nil {
+		log.Printf("[ERR] Error setting EMR configurations_json for cluster (%s): %s", d.Id(), err)
```

Comment:
sure

Reply:
@bflad when I returned the error message instead of logging, I came to know there is an error while setting the `configurations_json`. So I tried to decode the API `cluster. Configurations` output as json string using `json.Marshal` but I got an below error,

```
configurations_json:                                "[{\"Classification\":\"hadoop-env\",\"Configurations\":[{\"Classification\":\"export\",\"Configurations\":null,\"Properties\":{\"JAVA_HOME\":\"/usr/lib/jvm/java-1.8.0\"}}],\"Properties\":{}},{\"Classification\":\"spark-env\",\"Configurations\":[{\"Classification\":\"export\",\"Configurations\":null,\"Properties\":{\"JAVA_HOME\":\"/usr/lib/jvm/java-1.8.0\"}}],\"Properties\":{}}]" => "[{\"Classification\":\"hadoop-env\",\"Configurations\":[{\"Classification\":\"export\",\"Configurations\":[],\"Properties\":{\"JAVA_HOME\":\"/usr/lib/jvm/java-1.8.0\"}}],\"Properties\":{}},{\"Classification\":\"spark-env\",\"Configurations\":[{\"Classification\":\"export\",\"Configurations\":[],\"Properties\":{\"JAVA_HOME\":\"/usr/lib/jvm/java-1.8.0\"}}],\"Properties\":{}}]" (forces new resource)
```
since API returning nil (empty `[]*emr.Configuration`), `json.Marshal` converts it into `null` and causes the above issue. Looks like we can rewrite API output from `nil` to `[]*emr.Configuration{}` to fix this, but `emr.Configuration` type have field `Configurations` which value is again `[]*emr.Configuration` and it goes on. May be recursive function helps this case.
But I tried `jsonutil.BuildJSON` (https://godoc.org/github.com/aws/aws-sdk-go/private/protocol/json/jsonutil) to decode json, and it completely skips the field(`Configurations`), since it have nil value. so I think we can use this and we can update the document like, if `Configurations` is empty please completely `skip` the field instead of providing empty list `"Configurations": []` in input json same as first example (https://docs.aws.amazon.com/emr/latest/ReleaseGuide/emr-configure-apps.html).
Any thoughts?